lib/server/api/validators/timer.js

```diff
@@ -9,13 +9,22 @@ function typeOf(obj) {
   return type_exp.exec( Object.prototype.toString.call(obj) )[1];
 }
 
+const MAX_TIMEOUT_VALUE = Math.pow(2, 31) - 1;
+
 module.exports = function(data = {}, cb) {
-  if( isNaN(data.timeout) || data.timeout < 1 ) {
+  if (isNaN(data.timeout) || data.timeout < 1) {
     const err = new TypeError('timeout is required and must be a positive number');
     err.statusCode = 400;
     return setImmediate(cb, err);
   }
-  if(data.data) {
+
+  if (data.timeout > MAX_TIMEOUT_VALUE) {
+    const err = new TypeError(`timeout must be less than or equal to ${MAX_TIMEOUT_VALUE} milliseconds`);
```

Comment:
Would like to avoid string concatenation where possible
```suggestion
    const err = new TypeError(`timeout must be less than or equal to 2147483647 milliseconds`);
```